gunrock/app/color/color_enactor.cuh

```diff
@@ -165,62 +166,92 @@ struct ColorIterationLoop : public IterationLoopBase
         //    frontier.queue_index, frontier.queue_length,
         //    false, oprtr_parameters.stream, true));
 
-	} else {
+	}
+
+  else {
 
 	    auto color_op = [
-		graph,
+        use_jpl,
+		    graph,
 	    	colors,
 	    	rand,
 	    	iteration
 	    ] __host__ __device__ (VertexT *v_q, const SizeT &pos) {
 
-	    	VertexT v 		= v_q[pos];
-	    	SizeT start_edge 	= graph.CsrT::GetNeighborListOffset(v);
-	    	SizeT num_neighbors 	= graph.CsrT::GetNeighborListLength(v);
-
-	    	VertexT max = v;    // active max vertex
-		    VertexT min = v;    // active min vertex
-	    	ValueT temp = rand[v];
+        VertexT v 		= v_q[pos];
+        SizeT start_edge 	= graph.CsrT::GetNeighborListOffset(v);
+        SizeT num_neighbors 	= graph.CsrT::GetNeighborListLength(v);
+        ValueT temp = rand[v];
+
+        if(use_jpl) {
```

Comment:
This is fine for now, but ideally you would want a separate operator for JPL vs. the other implementation. It is much cleaner and easier to read what each implementation is doing, and then once you have declared all of them, just put an if condition around the actual run choosing which operator to run.